package/kedro_viz/data_access/managers.py

```diff
@@ -456,7 +456,9 @@ def create_modular_pipelines_tree_for_registered_pipeline(
                 or not node.belongs_to_pipeline(registered_pipeline_id)
             ):
                 continue
-            if not node.modular_pipelines or node_id in root_children_ids:
```

Comment:
The root modular pipeline is what we first see in the collapsed Kedro-viz pipeline. Initially, users should be able to view:

- Top-level modular pipelines
- Any tasks, datasets, or parameters that are not part of a modular pipeline
- Inputs and outputs of the top-level modular pipeline

The above if statement works as follows:
- It adds all nodes that don't belong to a modular pipeline to root
- It also adds inputs and outputs of a modular pipeline, including inputs and outputs of nested modular pipelines
-
The main distinction between inputs/outputs of top-level modular pipelines and inputs/outputs of nested modular pipelines is that the former doesn't have any associated modular pipeline. 

Therefore, we need to revise this if statement to include:
- Nodes that are not part of any modular pipeline
- Inputs and outputs of a modular pipeline (root_children_ids) that are also not part of the modular pipeline.